src/AppBundle/Command/DocsPublishCommand.php

```diff
@@ -8,47 +8,33 @@
 use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Console\Output\OutputInterface;
 
-class DocsPublishCommand extends ContainerAwareCommand
-{
-    protected function configure()
-    {
-        $this
-          ->setName('docs:publish')
-          ->setDescription('...')
-          ->addArgument('paths', InputArgument::IS_ARRAY,
-            'One or more book expressions ("book/lang/branch"). (Default: all)');
-    }
-
-    protected function execute(InputInterface $input, OutputInterface $output)
-    {
-        /** @var \AppBundle\BookLoader $books */
-        $books = $this->getContainer()->get('book.loader');
-
-        /** @var \AppBundle\Utils\Publisher $publisher */
-        $publisher = $this->getContainer()->get('publisher');
-
-        $rows = $books->findAsList();
-
-        if (empty($rows)) {
-            $output->writeln("<error>No books found</error>");
-        }
-
-        $rowKeys = $input->getArgument('paths') ? $input->getArgument('paths') : array_keys($rows);
-
-        foreach ($rowKeys as $rowKey) {
-            $row = $rows[$rowKey];
-            $output->writeln("");
-            $output->writeln(sprintf("Publish [%s/%s/%s] (from %s)",
-              $row['book'], $row['lang'], $row['branch'], $row['repo']));
-
-            $publisher->publish($row['book'], $row['lang'], $row['branch']);
-
-            foreach ($publisher->getMessages() as $message) {
-                $output->writeln($message['label'] . ': ' . $message['content']);
-            }
-            $publisher->clearMessages();
-
-        }
+class DocsPublishCommand extends ContainerAwareCommand {
+
+  protected function configure() {
+    $this
+      ->setName('docs:publish')
+      ->setDescription('Publish one or more books')
+      ->addArgument(
+          'identifiers',
+          InputArgument::IS_ARRAY,
+          'One or more book identifiers (e.g. "user/en/master"). Partial '
+          . 'identifiers are acceptable (e.g. "user/en" will publish all '
+          . 'English versions of the User Guide. If no identifiers are '
+          . 'specified, then all versions of all languages in all books will '
+          . 'be published.');
+  }
+
+  protected function execute(InputInterface $input, OutputInterface $output) {
+    /** @var \AppBundle\Utils\Publisher $publisher */
+    $publisher = $this->getContainer()->get('publisher');
+    $identifiers = $input->getArgument('identifiers');
```

Comment:
Also it's worth mentioning that the web interface will now allow publishing via "identifiers" as well, so you can go to `docs.civicrm.org/admin/publish/user` to publish all the languages and all the versions within that one book. But I built a [safeguard](https://github.com/seanmadsen/civicrm-docs/blob/refactor-model/src/AppBundle/Controller/PublishController.php#L35) into this page which prevents the publishing of everything (because publishing everything takes a long time).